Src/FluentAssertions/Primitives/DateTimeAssertions.cs

```diff
@@ -928,6 +928,35 @@ public AndConstraint<TAssertions> BeIn(DateTimeKind expectedKind, string because
         return new AndConstraint<TAssertions>((TAssertions)this);
     }
 
+    /// <summary>
+    /// Asserts that the <see cref="DateTime"/> do not represent a value in the <paramref name="unexpectedKind"/>.
+    /// </summary>
+    /// <param name="unexpectedKind">
+    /// The unexpected <see cref="DateTimeKind"/> that the current value must not represent.
+    /// </param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public AndConstraint<TAssertions> NotBeIn(DateTimeKind unexpectedKind, string because = "", params object[] becauseArgs)
+    {
+        Execute.Assertion
+            .BecauseOf(because, becauseArgs)
+            .WithExpectation("Did not expect {context:the date and time} to be in " + unexpectedKind + "{reason}")
+            .ForCondition(Subject.HasValue)
+            .FailWith(", but found a <null> DateTime.")
+            .Then
+            .ForCondition(Subject.Value.Kind != unexpectedKind)
+            .FailWith(", but it was.")
+            .Then
+            .ClearExpectation();
```

Comment:
We can make this safe against assertions wrapped in an `AssertionScope`

```suggestion
        Execute.Assertion
            .BecauseOf(because, becauseArgs)
            .WithExpectation("Did not expect {context:the date and time} to be in " + unexpectedKind + "{reason}")
            .Given(() => Subject)
            .ForCondition(subject => subject.HasValue)
            .FailWith(", but found a <null> DateTime.")
            .Then
            .ForCondition(subject => subject.Value.Kind != unexpectedKind)
            .FailWith(", but it was.")
            .Then
            .ClearExpectation();
```